src/libs/Network/SequentialQueue.js

```diff
@@ -24,20 +40,42 @@ let currentRequest = null;
  * @returns {Promise}
  */
 function process() {
-    const persistedRequests = PersistedRequests.getAll();
+    const persistedRequests = [...PersistedRequests.getAll()];
 
-    // This sanity check is also a recursion exit point
-    if (NetworkStore.isOffline() || _.isEmpty(persistedRequests)) {
+    // If we have no persisted requests or we are offline we don't want to make any requests so we return early
+    if (_.isEmpty(persistedRequests) || NetworkStore.isOffline()) {
```

Comment:
I think it's so that we can short circuit the OR statement when the persisted requests are empty, which will happen more often than the network being offline. cc @madmax330